org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/server/LanguageServerImpl.java

```diff
@@ -720,8 +745,10 @@ public CompletableFuture<Either<List<? extends SymbolInformation>, List<? extend
 	 * @since 2.20
 	 */
 	protected Either<List<? extends SymbolInformation>, List<? extends WorkspaceSymbol>> symbol(WorkspaceSymbolParams params, CancelIndicator cancelIndicator) {
-		return workspaceSymbolService.getSymbols(params.getQuery(), resourceAccess, workspaceManager.getIndex(),
+		Either<List<? extends WorkspaceSymbol>, List<? extends DocumentSymbol>> symbols = workspaceSymbolService.getSymbols(params.getQuery(), resourceAccess, workspaceManager.getIndex(),
 				cancelIndicator);
+		List<SymbolInformation> symbolsInformations = symbols.getLeft().stream().map(ws -> new SymbolInformation(ws.getName(), ws.getKind(), ws.getLocation().getLeft())).collect(Collectors.toList());
```

Comment:
i dont get why symbolinformation is still used here?
is this method still called somewhere?

Reply:
org.eclipse.xtext.ide.server.LanguageServerImpl.documentSymbol(DocumentSymbolParams, CancelIndicator)
should return DocumentSymbols, not Symbol informations.
same for org.eclipse.xtext.ide.server.LanguageServerImpl.symbol(WorkspaceSymbolParams)

Reply:
As org.eclipse.xtext.ide.server.LanguageServerImpl.documentSymbol(DocumentSymbolParams) overrides org.eclipse.lsp4j.services.TextDocumentService, and  so I converted WorkspaceSymbol to SymbolInformation. 

So I converted in org.eclipse.xtext.ide.server.LanguageServerImpl.documentSymbol(DocumentSymbolParams)
Will that be OK ?

Reply:
As org.eclipse.xtext.ide.server.LanguageServerImpl.documentSymbol(DocumentSymbolParams) overrides org.eclipse.lsp4j.services.TextDocumentService.documentSymbol(DocumentSymbolParams) and 
org.eclipse.xtext.ide.server.LanguageServerImpl.symbol(WorkspaceSymbolParams) overrides org.eclipse.lsp4j.services.WorkspaceService.symbol(WorkspaceSymbolParams)
and so I have to converted WorkspaceSymbol to SymbolInformation otherwise we will end-up with compiler error

Do you thinks should I converted in WorkspaceSymbol to SymbolInformation  in org.eclipse.xtext.ide.server.LanguageServerImpl.documentSymbol(DocumentSymbolParams) and org.eclipse.xtext.ide.server.LanguageServerImpl.symbol(WorkspaceSymbolParams)

Will that be OK ?

Reply:
I checked out your branch and still see the deprecated class instances created 
We should not do that

Reply:
=>
org.eclipse.xtext.ide.server.LanguageServerImpl.symbol(WorkspaceSymbolParams, CancelIndicator)
should return either.forright(List<WorkspaceSymbol>

=>
org.eclipse.xtext.ide.server.LanguageServerImpl.documentSymbol(DocumentSymbolParams, CancelIndicator)
should return  List<Either.forRight(DocumentSymbol)

Reply:
Yes its because of SymbolInformation, although they have marked it deprecated but still use it in org.eclipse.lsp4j.services.TextDocumentService.documentSymbol(DocumentSymbolParams) which returns CompletableFuture<List<Either<SymbolInformation, DocumentSymbol>>>

Is there a way we can modify or update LSP4J project ?

Reply:
org.eclipse.xtext.ide.server.symbol.WorkspaceSymbolService.getSymbols(String, IResourceAccess, IResourceDescriptions, CancelIndicator) might also need to be changed to produce list workspacesymbol
org.eclipse.xtext.ide.server.symbol.DocumentSymbolService.getSymbols(XtextResource, CancelIndicator) might need produce Document Symbols only

Reply:
similar for org.eclipse.xtext.ide.server.symbol.HierarchicalDocumentSymbolService.getSymbols(XtextResource, CancelIndicator)

Reply:
The only place the deprecated classes should stay is in the method signatures of the methods override in lsp4j
But the return values should be either. Fortgethingthatisnotreprecated

Reply:
Marked PR is WIP as making org.eclipse.xtext.ide.server.symbol.DocumentSymbolService.getSymbols(XtextResource, CancelIndicator) returns DocumentSymbol fails IDE tests. 

HierarchicalDocumentSymbolService returns the DocumetSymbol in right where as DocumentSymbolService returns WorkspaceSymbol in left side (earlier it was SymbolInformation) -- In my view WorkspaceSymbol is better in replacing  SymbolInformation as it has location object where are DocumentSymbol doesn't have that.

Will go though org.eclipse.xtext.testing.AbstractLanguageServerTest.testDocumentSymbol((DocumentSymbolConfiguration)=>void) to check if how can modify it.